controllers/drplacementcontrol_controller_test.go

```diff
@@ -1658,101 +1645,38 @@ func deleteDRPolicySync() {
 }
 
 func fenceCluster(cluster string, manual bool) {
-	localRetries := 0
-	for localRetries < updateRetries {
-		latestDRCluster := getLatestDRCluster(cluster)
-		if manual {
-			latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateManuallyFenced
-		} else {
-			latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateFenced
-		}
-
-		err := k8sClient.Update(context.TODO(), latestDRCluster)
-		if errors.IsConflict(err) {
-			localRetries++
-
-			continue
-		}
-
-		Expect(err).NotTo(HaveOccurred())
-
-		break
+	latestDRCluster := getLatestDRCluster(cluster)
+	if manual {
+		latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateManuallyFenced
+	} else {
+		latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateFenced
 	}
 
-	Eventually(func() bool {
-		latestDRCluster := getLatestDRCluster(cluster)
-
-		drClusterFencedCondition := getDRClusterCondition(&latestDRCluster.Status,
-			rmn.DRClusterConditionTypeFenced)
-		if drClusterFencedCondition == nil {
-			return false
-		}
-
-		if drClusterFencedCondition.ObservedGeneration != latestDRCluster.Generation {
-			return false
-		}
-
-		return drClusterFencedCondition.Status == metav1.ConditionTrue
-	}, timeout, interval).Should(BeTrue(), "failed to update DRCluster on time")
+	latestDRCluster = updateDRClusterParameters(latestDRCluster)
+	drclusterConditionExpectEventually(latestDRCluster, false, metav1.ConditionTrue,
+		Equal(controllers.DRClusterConditionReasonFenced), Ignore(),
+		rmn.DRClusterConditionTypeFenced)
 }
 
 func unfenceCluster(cluster string, manual bool) {
-	localRetries := 0
-	for localRetries < updateRetries {
-		latestDRCluster := getLatestDRCluster(cluster)
-		if manual {
-			latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateManuallyUnfenced
-		} else {
-			latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateUnfenced
-		}
-
-		err := k8sClient.Update(context.TODO(), latestDRCluster)
-		if errors.IsConflict(err) {
-			localRetries++
-
-			continue
-		}
-
-		Expect(err).NotTo(HaveOccurred())
-
-		break
+	latestDRCluster := getLatestDRCluster(cluster)
+	if manual {
+		latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateManuallyUnfenced
+	} else {
+		latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateUnfenced
 	}
 
-	Eventually(func() bool {
-		latestDRCluster := getLatestDRCluster(cluster)
-
-		drClusterFencedCondition := getDRClusterCondition(&latestDRCluster.Status,
-			rmn.DRClusterConditionTypeFenced)
-		if drClusterFencedCondition == nil {
-			return false
-		}
-
-		if drClusterFencedCondition.ObservedGeneration != latestDRCluster.Generation {
-			return false
-		}
-
-		return drClusterFencedCondition.Status == metav1.ConditionFalse
-	}, timeout, interval).Should(BeTrue(), "failed to update DRCluster on time")
```

Comment:
...but I'm not sure about this change - replacing the `Eventually()` block with a call to `drclusterConditionExpectEventually()`.

Maybe it is correct, but this is exactly what we should not do - mix unrelated changes in the same commit. This makes it much harder to understand the change both during review of when trying to fix an issue later.

Reply:
The change itself is correct, we are doing the same in other places in code (for example, in drclustercontroller_test.go), but it can be separated into another commit

Reply:
Move to a separate commit